frontend_csv/src/scanner/scanner.rs

```diff
@@ -1,13 +1,202 @@
 use super::token::CSVToken;
-
+use super::token::CSVTokenType;
 /// Core implementation of the scanner
+use std::collections::HashMap;
+
 pub struct Scanner {
-    pos: usize,
     line: u64,
-    tokens: Vec<CSVToken>,
-    identifier_symbols: Vec<char>,
+    keywords: HashMap<String, CSVToken>,
 }
 
 impl Scanner {
-    fn scan(&mut self, _symbols: &Vec<char>) {}
+    pub fn new(line: u64) -> Scanner {
```

Comment:
```suggestion
    pub fn new() -> Scanner {
```

The line is always 1